### PR TITLE
Support for ignoring tests when running valgrind

### PR DESCRIPTION
**Pull request by wiredfool**
Before we can add valgrind as an experimental docker test target, we need to have all of the tests passing or known ignored.

This adds:

* a pytest configuration for ignoring the error when pytest-valgrind isn't installed. 
* Ignores the tiff_additional_metadata test, because we're sending some illogical values to it. 

WIP -- there are more tests to investigate.

### CODE REVIEW

File: Tests/conftest.py [hunk 1]
```diff
@@ -10,3 +10,18 @@ def pytest_report_header(config):
...
+    # Ensure that the mark is defined even in cases where pytest-valgrind isn't installed
+
+    import pytest
+    import warnings
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        try:
+            getattr(pytest.mark, "valgrind_known_error")
+        except:
```
Comment (hugovk) [inline Tests/conftest.py:25]: ?
```suggestion
        except AttributeError:
```

File: Tests/test_file_libtiff.py [hunk 2]
```diff
@@ -185,6 +185,7 @@ def test_write_metadata(self, tmp_path):
             for field in requested_fields:
                 assert field in reloaded, f"{field} not in metadata"
 
+    @pytest.mark.valgrind_known_error(reason="Known Invalid Metadata")
```
Comment (hugovk) [inline Tests/test_file_libtiff.py:188]:
```suggestion
    @pytest.mark.valgrind_known_error(reason="Known invalid metadata")
```

File: Tests/conftest.py [hunk 3]
```diff
@@ -10,3 +10,18 @@ def pytest_report_header(config):
             return out.getvalue()
     except Exception as e:
         return f"pytest_report_header failed: {e}"
+
+def pytest_configure(config):
+    # We're marking some tests to ignore valgrind errors and XFAIL them.
+    # Ensure that the mark is defined even in cases where pytest-valgrind isn't installed
+
+    import pytest
+    import warnings
```
Comment (hugovk) [inline Tests/conftest.py:19]: These could go at the top of the file.

File: Tests/conftest.py [hunk 4]
```diff
@@ -10,3 +10,18 @@ def pytest_report_header(config):
             return out.getvalue()
     except Exception as e:
         return f"pytest_report_header failed: {e}"
+
```
Comment (radarhere) [inline Tests/conftest.py:13]:
```suggestion


```
Lint fix

File: Tests/conftest.py [hunk 5]
```diff
@@ -10,3 +10,18 @@ def pytest_report_header(config):
...
+    import pytest
+    import warnings
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        try:
+            getattr(pytest.mark, "valgrind_known_error")
+        except:
+            config.addinivalue_line("markers",
+                                    "valgrind_known_error: Tests that have known issues with valgrind")
```
Comment (radarhere) [inline Tests/conftest.py:27]:
```suggestion
            config.addinivalue_line(
                "markers",
                "valgrind_known_error: Tests that have known issues with valgrind"
            )
```
Lint fix

File: Tests/conftest.py (hunk 1 above)
Comment (wiredfool) [reply Tests/conftest.py:28]: I don't think it's an attribute error, as it's catching a warning thrown by pytest which hooks the getattr to specifically avoid an attribute error.

**Pull request merged**